topsort.go

```diff
@@ -148,6 +148,10 @@ func sortNodes(nodes []*Node) (ret []*Node, err error) {
 		}
 		ret = append(ret, generation...)
 	}
+	/* Reset Marked status of nodes now that we're done */
+	for _, node := range nodes {
+		node.Marked = false
+	}
```

Comment:
Maybe it would make more sense to do this at the start of this function instead of the end?  Not sure.